### PR TITLE
Quick-fix for d.correlate.py

### PR DESCRIPTION
**Pull request by NikosAlexandris**
- Use open() instead of file()
    - Use a bytes-like object for ofile.write()

### CODE REVIEW

File: scripts/d.correlate/d.correlate.py [hunk 1]
```diff
@@ -98,12 +98,12 @@ def main():
...
                 ofile = p.stdin
 
-                ifile = file(tmpfile, 'r')
+                ifile = open(tmpfile, 'r')
                 for l in ifile:
                     f = l.rstrip('\r\n').split(' ')
                     x = float(f[0])
                     y = float(f[1])
-                    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
+                    ofile.write(b"icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```
Comment (petrasovaa) [inline scripts/d.correlate/d.correlate.py:106]: I haven't actually test your changes, but this seems like it won't work in Python 3, you opened the file in text mode and you write there bytestring, so you should be getting:
`TypeError: write() argument must be str, not bytes`.

**Review by petrasovaa — CHANGES_REQUESTED** [review]

File: scripts/d.correlate/d.correlate.py (hunk 1 above)
Comment (NikosAlexandris) [reply scripts/d.correlate/d.correlate.py:106]: Ciao Anna. I just changed `file()` to `open()`. Then,
```
python --version
Python 3.8.5
```
and
```
→ d.correlate map=S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_ASCENDING_20190610_20190617_median_VV,S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_DESCENDING_20190610_20190617_median_VV
 100%
Traceback (most recent call last):
  File "/osgeo/grass/dist.x86_64-pc-linux-gnu/scripts/d.correlate", line 119, in <module>
    sys.exit(main())
  File "/osgeo/grass/dist.x86_64-pc-linux-gnu/scripts/d.correlate", line 106, in main
    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
TypeError: a bytes-like object is required, not 'str'
```
then, changing from
```
                    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```
to
```
                    ofile.write(b"icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```
works here
```
→ d.correlate map=S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_ASCENDING_20190610_20190617_median_VV,S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_DESCENDING_20190610_20190617_median_VV
 100%
```
and I get an image output too.

Comment (petrasovaa) [reply scripts/d.correlate/d.correlate.py:106]: Sorry, I overlooked `ofile` is redefined on line 99, that makes sense now.

**Review by petrasovaa — APPROVED** [review]

**Comment by neteler** [comment]
Is this a bugfix and needs to be backported or a nice-to-have?

**Comment by NikosAlexandris** [comment]
Yes, a fix (Python 2 vs Python 3). A quick-one. The module seems not to be used by anyone.
It deserves an overall review, in my humble view, if it's meant to stay.

**Comment by wenzeslaus** [comment]
I agree @NikosAlexandris, a review/improvement is needed. It can always go to Addons if we decide it is not core-worthy. I used (and wrote) [r.scatterplot](https://grass.osgeo.org/grass78/manuals/addons/r.scatterplot.html) which is much more flexible (3D or computed, not visual, density), but it requires some extra steps and it may be a strange workflow for some. _d.correlate_ is just simpler and obviously integrated with the rest of rendering. And it could stay simple even with some additions. It is not included in the GUI which might be one of the reasons it is being overlooked.

**Pull request merged**